dot/parachain/types/types.go

```diff
@@ -477,6 +477,11 @@ func (cr CandidateReceipt) Hash() (common.Hash, error) {
 	return common.Blake2bHash(bytes)
 }
 
+type CandidateReceiptV2 struct {
+	Descriptor      *CandidateDescriptorV2
+	CommitmentsHash common.Hash `scale:"2"`
+}
+
```

Comment:
Did you ensure the scale encoding/decoding?
`Descriptor` is not an option in Rust, so we should not use a pointer in order to ensure proper encoding/decoding.
```
#[derive(PartialEq, Eq, Clone, Encode, Decode, DecodeWithMemTracking, TypeInfo, RuntimeDebug)]
#[cfg_attr(feature = "std", derive(Hash))]
pub struct CandidateReceiptV2<H = Hash> {
	/// The descriptor of the candidate.
	pub descriptor: CandidateDescriptorV2<H>,
	/// The hash of the encoded commitments made as a result of candidate execution.
	pub commitments_hash: Hash,
}
```

Reply:
Also, If you want to have scale tags, add it to both or remove if you want. currently you have scale tag only for second field and not for the first field.

Reply:
Given that you've made PR adding candidate receipt v2 I will wait the merge and bring your changes here.